rpt.js

```diff
@@ -6,6 +6,23 @@ var once = require('once')
 var readdir = require('readdir-scoped-modules')
 var debug = require('debuglog')('rpt')
 
+function asyncForEach (items, todo, done) {
+  var remaining = items.length
+  if (remaining === 0) return done()
+  var seenErr
+  items.forEach(function (item) {
+    todo(item, handleComplete)
+  })
+  function handleComplete (err) {
+    if (seenErr) return
+    if (err) {
+      seenErr = true
+      return done(err)
+    }
+    if (--remaining === 0) done()
```

Comment:
my recently-inflamed paranoia around race conditions prompts me to ask if this would be better replaced with `--remaining < 1`